### PR TITLE
Add queue array impl in C

### PR DESCRIPTION
**Pull request by Gonglja**
If this PR is related to coding or code translation, please fill out the checklist.

- [ ] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [ ] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [ ] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

**Comment by krahets** [comment]
Could you copy&paste the console outputs to this PR? Thanks!

**Comment by Gonglja** [comment]
Ok, no problem, the following is
```shell
C:\Windows\system32\wsl.exe --distribution Ubuntu-22.04 --exec /bin/zsh -c "cd /mnt/d/03Data/01Git/hello-algo/codes/c/cmake-build-debug/chapter_stack_and_queue && /mnt/d/03Data/01Git/hello-algo/codes/c/cmake-build-debug/chapter_stack_and_queue/array_queue"
队列 queue = [1, 3, 2, 5, 4]
队首元素 peek = 1
出队元素 poll = 1，出队后 queue = [3, 2, 5, 4]
队列长度 size = 4
队列是否为空 = false
第 0 轮入队 + 出队后 queue = [2, 5, 4, 0]
第 1 轮入队 + 出队后 queue = [5, 4, 0, 1]
第 2 轮入队 + 出队后 queue = [4, 0, 1, 2]
第 3 轮入队 + 出队后 queue = [0, 1, 2, 3]
第 4 轮入队 + 出队后 queue = [1, 2, 3, 4]
第 5 轮入队 + 出队后 queue = [2, 3, 4, 5]
第 6 轮入队 + 出队后 queue = [3, 4, 5, 6]
第 7 轮入队 + 出队后 queue = [4, 5, 6, 7]
第 8 轮入队 + 出队后 queue = [5, 6, 7, 8]
第 9 轮入队 + 出队后 queue = [6, 7, 8, 9]

Process finished with exit code 0

```

File: codes/c/chapter_stack_and_queue/array_queue.c [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        arr[i] = queue->nums[j % cap];
+    }
+    printArray(arr, siz);
+}
+
+
+/* Driver Code */
+int main() {
+  /* 初始化队列 */
+  int capacity = 10;
```
Comment (krahets) [inline codes/c/chapter_stack_and_queue/array_queue.c:90]: Please use 4 spaces for indentation

File: codes/c/chapter_stack_and_queue/array_queue.c [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    queue->front = (queue->front + 1) % capacity(queue);
+}
+
+/* 打印基于环形数组形成的队列 */
+void printArrayQueue(ArrayQueue *queue) {
+    int siz = size(queue);
+    int cap = capacity(queue);
+    int arr[siz];
+    // 拷贝
+    for (int i=0, j=queue->front; i<siz; i++, j++) {
```
Comment (krahets) [inline codes/c/chapter_stack_and_queue/array_queue.c:80]: Please format the code

`for (int i = 0, j = queue->front; i < siz; i++, j++)`

Recommend using a code formatting tool.

File: codes/c/chapter_stack_and_queue/array_queue.c [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+}
+
+/* 判断队列是否为空 */
+bool empty(ArrayQueue *queue) {
+    return queue->rear - queue->front == 0;
+}
+
+/* 访问队首元素 */
+int peek(ArrayQueue *queue) {
+  assert(size(queue)!=0);
```
Comment (krahets) [inline codes/c/chapter_stack_and_queue/array_queue.c:54]: `assert(size(queue) != 0);`

Please check the format of other codes.

**Review by krahets — COMMENTED** [review]
Thanks! Please address the comments.

File: codes/c/chapter_stack_and_queue/array_queue.c (hunk 1 above)
Comment (Gonglja) [reply codes/c/chapter_stack_and_queue/array_queue.c:90]: > Please use 4 spaces for indentation

Ok, I reformatted some code, sorry for the different results due to just cutting to clion using different tools to format differently

**Review by krahets — APPROVED** [review]
Thanks! Now merge it.

**Pull request merged**